diffsync/store/__init__.py

```diff
@@ -188,6 +206,34 @@ def update_or_instantiate(
 
         return obj, created
 
+    def update_or_instantiate_model(self, obj: "DiffSyncModel") -> Tuple["DiffSyncModel", bool]:
+        """Attempt to update an existing object with provided ids/attrs or instantiate obj.
+
+        Args:
+            instance: An instance of the DiffSyncModel to update or create.
+
+        Returns:
+            Provides the existing or new object and whether it was created or not.
+        """
+        model = obj.get_type()
+        ids = obj.get_unique_id()
+        attrs = obj.get_attrs()
+
+        created = False
+        try:
+            obj = self.get(model=model, identifier=ids)
+        except ObjectNotFound:
+            # Add the object to diffsync adapter
+            self.add(obj=obj)
+            created = True
+
+        # Update existing obj with attrs
+        for attr, value in attrs.items():
+            if getattr(obj, attr) != value:
```

Comment:
I believe we can omit this check to make it simpler. Just `setattr` always, if its the same the outcome is also the same.